pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -212,12 +215,32 @@ func MergeRuntimeContainers(runtimeContainer *v1.Container, predictorContainer *
 		mergedContainer.Name = runtimeContainerName
 	}
 
-	// Strategic merge patch will replace args but more useful behaviour here is to concatenate
-	mergedContainer.Args = append(append([]string{}, runtimeContainer.Args...), predictorContainer.Args...)
+	// Only concatenate args when there's no overlap. See discussions in https://github.com/kserve/kserve/pull/3444
+	if len(findIntersection(runtimeContainer.Args, predictorContainer.Args)) == 0 {
+		mergedContainer.Args = append(append([]string{}, runtimeContainer.Args...), predictorContainer.Args...)
+		log.Info("Concatenated args from runtime and predictor containers since there's no overlap")
+	} else {
+		log.Info("Predictor container args will override the runtime container args since an overlap is found")
```

Comment:
This might not be backwards compatible as there could be cases when you need take args from the serving runtime while only overriding the `--model_name` arg.